### PR TITLE
WT-10106 dereference null pointer

### PR DESCRIPTION
**Pull request by adelinexchen**
Assert error if pointer is ever null

### CODE REVIEW

File: test/utility/tiered.c [hunk 1]
```diff
@@ -63,6 +63,7 @@ testutil_tiered_sleep(TEST_OPTS *opts, WT_SESSION *session, uint64_t seconds, bo
 {
     uint64_t now, wake_time;
     bool do_flush;
+    WT_ASSERT(session, do_flush_tier != NULL);
 
     now = testutil_time_us(session);
     wake_time = now + WT_MILLION * seconds;
```
Comment (Sean04) [inline test/utility/tiered.c:69]: We can remove `do_flush_tier != NULL` from the if check below now too.

File: test/utility/tiered.c [hunk 2]
```diff
@@ -72,6 +73,7 @@ testutil_tiered_sleep(TEST_OPTS *opts, WT_SESSION *session, uint64_t seconds, bo
         wake_time = opts->tiered_flush_next_us;
         do_flush = true;
```
Comment (Sean04) [inline test/utility/tiered.c:74]: This isn't part of your change but `do_flush` should be a pointer here.

**Review by Sean04 — APPROVED** [review]
LGTM :)

**Review by marcbutler — APPROVED** [review]
LGTM

**Comment by adelinexchen** [comment]
evergreen merge

**Pull request merged**